src/template.yml

```diff
@@ -30,48 +30,69 @@ Parameters:
   CrossAccountAccessRoleName:
     Type: String
     Default: OrganizationAccountAccessRole
+    AllowedPattern: "\\A[a-zA-Z0-9_+=,.@\\-]+\\Z"
 
   MainNotificationEndpoint:
     Type: String
     Default: ""
-    Description: "Example -> jane@example.com"
+    Description: >-
+      Example -> jane@example.com. (Only required when installing ADF for the
+      first time.)
 
   DeploymentAccountName:
     Type: String
     Default: ""
-    Description: "The name of the new or existing deployment account"
+    Description: >-
+      The name of the new or existing deployment account. (Only required when
+      installing ADF for the first time.)
 
   DeploymentAccountEmailAddress:
     Type: String
     Default: ""
-    Description: "The email address for the new or existing deployment account"
+    Description: >-
+      The email address for the new or existing deployment account. (Only required when
+      installing ADF for the first time.)
 
   DeploymentAccountAlias:
     Type: String
     Default: ""
-    Description: "(Optional) Example -> companyname-deployment (must be globally unique)"
+    AllowedPattern: "\\A([a-z0-9]([a-z0-9]|-(?!-))*[a-z0-9])?\\Z"
```

Comment:
[Documentation](https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateAccountAlias.html) uses a different Regex (see capture group start `(`)

`[a-z0-9](([a-z0-9]|-(?!-))*[a-z0-9])?`

Or is the capture group different because we allow an empty string to be passed?

Reply:
Well this is minor, but by moving the capture group you no longer allow single char aliases (that are allowed in the original regex, see first part `[a-z0-9]`). 

Instead you could just add another like this:

`([a-z0-9](([a-z0-9]|-(?!-))*[a-z0-9])?)?`

Reply:
I would be surprised if those aliases are still available. But happy to change it 😄